### ISSUE TITLE
splitValue occasionally errors out

### ISSUE THREAD

**Issue by adamziel** [WordPress/gutenberg#34174]
### Description

The following error occasionally happens in the post editor:

```
TypeError: Object is not iterable (cannot read property Symbol(Symbol.iterator))
```

The trace points back to this line:

https://github.com/WordPress/gutenberg/blob/26f72abebd9d2c57bdc9ac62a41def330e4e13a7/packages/block-editor/src/components/rich-text/split-value.js#L24

which is called somewhere here in useEnter:

https://github.com/WordPress/gutenberg/blob/26f72abebd9d2c57bdc9ac62a41def330e4e13a7/packages/block-editor/src/components/rich-text/use-enter.js#L102-L107

I suppose something is wrong with `propsRef.current.value`

cc @ellatrix 

### Step-by-step reproduction instructions

I don't have the exact reproduction instructions unfortunately, but there are no other plugins or themes in the mix.

### Pre-checks

- [x] I have searched the existing issues.
- [x] I have tested with all plugins deactivated except Gutenberg.

**Comment by adamziel** [comment]
This seems to only be possible if `start` or `end` is undefined here:

https://github.com/WordPress/gutenberg/blob/7b0736c5a97f770cc931616bd0d5c0e446584590/packages/rich-text/src/split.js#L57-L65

And `typeof string !== 'string'` here:
https://github.com/WordPress/gutenberg/blob/7b0736c5a97f770cc931616bd0d5c0e446584590/packages/rich-text/src/split.js#L19-L22

**Comment by ellatrix** [comment]
I think this was fixed by #39822. Let's reopen if not :)
